src/api/users.ts

```diff
@@ -36,4 +42,75 @@ usersRouter.post("/", async (req, res, next) => {
     }
 })
 
-export default usersRouter;
\ No newline at end of file
+usersRouter.post("/register", async (req, res, next) => {
+    try {
+        const { email, username, password } = req.body;
+        bcrypt.hash(password, SALT_ROUNDS, async function(err: Error | undefined, hash: string) {
+            const user = await prisma.user.create({
+            data: {
+                email,
+                username,
+                password: hash
+            }
+        })
+        // JSON Web Token returned to client
+        const token = jwt.sign({
+            username: user.username,
+            id: user.id,
+        }, ACCESS_TOKEN_SECRET);
+        
+        res.send({
+            token,
+            user: {
+                email: user.email,
+                username: user.username
+            }
+        });
+    })
+    } catch (e) {
+        next(e)
+    }
+})
+
+usersRouter.post("/login", async (req, res, next) => {
+    try {
+        // throw new Error("An error");
+        const { email, password } = req.body;
+
+        const user = await prisma.user.findUniqueOrThrow({
+            where: {
+                email: email
+            }
+        })
+
+        bcrypt.compare(password, user.password, async function (err: Error | undefined, result: boolean) {
```

Comment:
nit: As mentioned, prefer async operation here to avoid callbacks. Not blocking on this tho.